### PR TITLE
update validation for separation

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
[Ticket](https://metaphase.atlassian.net/browse/TM-5448)

Makes TOD, ETA, and TED not required for separation leg.

### CODE REVIEW

File: talentmap_api/fsbid/services/agenda_item_validator.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def validate_individual_leg(leg):
             'valid': True,
             'errorMessage': ''
         },
-        'tod': validate_tod(leg['tod'], leg['tod_months'], leg['tod_long_desc']),
+        'tod': validate_tod(leg.get('is_separation') or False, leg['tod'], leg['tod_months'], leg['tod_long_desc']),
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda_item_validator.py:106]:
```suggestion
        'tod': validate_tod(leg.get('is_separation', False), leg['tod'], leg['tod_months'], leg['tod_long_desc']),
```

Comment (M00staff) [reply talentmap_api/fsbid/services/agenda_item_validator.py:106]: i THINK it works either way - but i touched this line as well and have it like that 🤷

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/agenda_item_validator.py:106]: it'll only default to False if it doesnt find the key, but if it does and the key is null, we wont get the False.

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/agenda_item_validator.py:106]: *value is null

File: talentmap_api/fsbid/services/agenda_item_validator.py [hunk 2]
```diff
@@ -118,14 +118,14 @@ def validate_individual_leg(leg):
         }
     }
 
-    # Leg - must have ETA
-    if not leg['eta']:
+    # Leg - must have ETA, if not separation
+    if not leg['eta'] and not leg.get('is_separation') or False:
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/agenda_item_validator.py:122]: updated after confirming or False doesnt mess with the first if not eval

**Review by M00staff — APPROVED** [review]

**Pull request merged**